src/main.go

```diff
@@ -145,18 +145,29 @@ func setHashJoinConcurrency(db *sql.DB) {
 	}
 }
 
+// isTiDB returns true if the DB is confirmed to be TiDB
+func isTiDB(db *sql.DB) bool {
+	if _, err := db.Exec("SELECT tidb_version()"); err != nil {
```

Comment:
Could this be done once per db only? I.e. checked once and then cached.

Reply:
It could :-) But this follows some existing logic where it does repeat some of these steps in setting various tidb settings which could also be cached. I didn't want to refactor it, but it is indeed redundant.